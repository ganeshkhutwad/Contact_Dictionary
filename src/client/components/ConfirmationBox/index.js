export { default as ConfirmationBox } from './ConfirmationBox';