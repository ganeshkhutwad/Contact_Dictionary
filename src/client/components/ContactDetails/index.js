export { default as ContactDetails } from './ContactDetails';