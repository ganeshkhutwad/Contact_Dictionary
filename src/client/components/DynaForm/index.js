export { default as DynaForm } from './DynaForm';
