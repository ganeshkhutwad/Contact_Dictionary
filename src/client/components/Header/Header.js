/**
@author Ganesh Khutwad
*/
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import AccountCircle from '@material-ui/icons/AccountCircle';

const styles = {
  root: {
    flexGrow: 1
  },
  flex: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
};

class Header extends Component {
  state = {
    auth: true
  };

  render() {
    const { classes, heading } = this.props;
    const { auth } = this.state;

    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
            <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
              <MenuIcon />
            </IconButton>
            <Typography variant="title" color="inherit" className={classes.flex}>
              {heading}
            </Typography>
            {auth && (
              <div>
                <IconButton
                  aria-haspopup="false"
                  color="inherit"
                >
                  <AccountCircle />
                </IconButton>
              </div>
            )}
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

Header.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Header);