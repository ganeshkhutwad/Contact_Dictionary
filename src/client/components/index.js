export { Header } from './Header';
export { ContactList } from './ContactList';
export { ContactDetails } from './ContactDetails';
export { AddButton } from './AddButton';
