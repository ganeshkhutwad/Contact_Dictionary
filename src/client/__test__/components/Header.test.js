/**
Test `AddButton` Presentational Component.
@author Ganesh Khutwad
 */
import React from 'react';
import { Header } from '../../components/Header';

const classes = { };

describe('`Header` Tests', () => {
    it('`Header` Snapshot', () => {
        const headerWrapper = shallow(<Header classes={classes} />);
        expect(headerWrapper).toMatchSnapshot();
    });
});
