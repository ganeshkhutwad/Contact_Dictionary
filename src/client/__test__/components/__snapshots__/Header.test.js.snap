// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Header\` Tests \`Header\` Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Header)
    classes={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "flex": "Header-flex-2",
        "menuButton": "Header-menuButton-3",
        "root": "Header-root-1",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "flex": "Header-flex-2",
          "menuButton": "Header-menuButton-3",
          "root": "Header-root-1",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
